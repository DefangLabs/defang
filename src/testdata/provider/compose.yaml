services:
  chat:
    image: my-chat-app
    # image: curlimages/curl
    # command:
    #   - "sh"
    #   - "-c"
    #   - "curl -sf $${AI_RUNNER_URL}chat/completions -H 'Content-Type: application/json' -d '{\"model\":\"anthropic.claude-3-5-sonnet-20241022-v2:0\",\"messages\":[{\"role\":\"user\",\"content\":\"Hello, world!\"}]}'; sleep 5"
    depends_on:
      - ai_runner

  ai_runner:
    provider:
      type: model
      options:
        model: ai/smollm2
    environment:
      DEBUG: "true"
    # x-defang-llm: true
