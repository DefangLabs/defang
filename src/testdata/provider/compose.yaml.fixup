{
  "ai_runner": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "OPENAI_API_KEY": ""
    },
    "image": "defangio/openai-access-gateway",
    "networks": {
      "model_provider_private": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 80,
        "protocol": "tcp"
      }
    ]
  },
  "chat": {
    "command": null,
    "depends_on": {
      "ai_runner": {
        "condition": "service_started",
        "required": true
      }
    },
    "entrypoint": null,
    "environment": {
      "AI_RUNNER_MODEL": "ai/smollm2",
      "AI_RUNNER_URL": "http://ai-runner/api/v1/"
    },
    "image": "my-chat-app",
    "networks": {
      "default": null,
      "model_provider_private": null
    }
  }
}