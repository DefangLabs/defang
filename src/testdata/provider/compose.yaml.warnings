 ! service "ai_runner": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "chat": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
