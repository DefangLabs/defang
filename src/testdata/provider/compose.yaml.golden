name: provider
services:
  ai_runner:
    provider:
      type: model
      options:
        model: ai/smollm2
    networks:
      default: null
  chat:
    depends_on:
      ai_runner:
        condition: service_started
        required: true
    image: my-chat-app
    networks:
      default: null
networks:
  default:
    name: provider_default
